src/Features/Core/Portable/SimplifyInterpolation/Helpers.cs

```diff
@@ -76,7 +76,9 @@ private static IOperation Unwrap(IOperation expression)
                 !syntaxFacts.IsBaseExpression(invocation.Instance.Syntax))
             {
                 if (invocation.Arguments.Length == 1 &&
-                    invocation.Arguments[0].Value is ILiteralOperation { ConstantValue: { HasValue: true, Value: string value } } literal)
+                    invocation.Arguments[0].Value is ILiteralOperation { ConstantValue: { HasValue: true, Value: string value } } literal &&
+                    invocation.SemanticModel.Compilation.GetTypeByMetadataName("System.IFormattable") is { } systemIFormattable &&
+                    invocation.Instance.Type.Implements(systemIFormattable))
```

Comment:
we likely want to ensure that this .ToString *is* the impl of the IFormattable.ToString right?  you can use .FindImplemetnationForInterfaceMember to validate that.  It would possibly be good to have a test where these are different as well (i.e. an explicit impl of htat interface).

Reply:
@CyrusNajmabadi It can't be. IFormattable only declares ToString(string, IFormatProvider). That's why I didn't do anything about it at first.